apps/emqx_durable_storage/src/emqx_ds.erl

```diff
@@ -125,6 +128,9 @@
 -callback make_iterator(db(), ds_specific_stream(), topic_filter(), time()) ->
     make_iterator_result(ds_specific_iterator()).
 
+-callback make_iterator_from_key(db(), ds_specific_iterator(), message_key()) ->
```

Comment:
Q: why does this function takes iterator instead of stream as the second argument?

Reply:
Because the use case here is for the cache to construct a new iterator based on the last seen key.  Otherwise, we would need all the info to create an iterator from scratch: https://github.com/emqx/emqx/pull/12125#discussion_r1419260458

And when using `next`, the arguments do not contain the original arguments, but the iterators internally do.

Reply:
I see, it makes sense. But a more appropriate name for the function I guess is `update_iterator`?